slmsuite/hardware/slms/simulated.py

```diff
@@ -0,0 +1,79 @@
+"""
+Simulated SLM. 
+"""
+
+import numpy as np
+from .slm import SLM
+from slmsuite.misc.fitfunctions import gaussian2d
+
+class SimulatedSLM(SLM):
+    """
+    A simulated SLM to emmulate physical artifacts of actual SLMs.
+
+    Attributes
+    ----------
+    resolution : tuple
+        (width, height) of the SLM in pixels.
+    phase_offset : ndarray
+        Phase delay array (radians) emmulating the result of physical SLM curvature. 
```

Comment:
Maybe consider naming this `phase_aberration`?

Hmm, maybe these should parallel the `measured_amplitude` and `phase_correction` counterparts. Something like `simulated_amplitude` and `phase_aberration`?

I also don't really like `measured_amplitude` at this point. We might consider renaming to match package style and uniformity.

Reply:
Moved all source attributes to a dictionary (slm.source). "amplitude" and "phase" keywords cover the measured profile, "amplitude_sim" and "phase_sim" are for the user-defined simulated source.